src/item.h

```diff
@@ -1229,6 +1229,15 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
             /** flags change behavior of gun mode and are **not** equivalent to item flags */
             std::set<std::string> flags;
 
+            gun_mode( const std::string &mode = "", item *target = nullptr, int qty = 0 ) :
```

Comment:
These should be provided as default values for members themselves. Two constructors additional constructors to satisfy a compiler bug seems excessive